### PR TITLE
refactor(ctl): unregister risectl node after use

### PR DESCRIPTION
**Pull request by zwang28**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Fix 2 TODOs:

- [Use proper host addr for risectl node](https://github.com/risingwavelabs/risingwave/blob/ab326f32de70706eba0d92b8688a6f01e0d2dffb/src/ctl/src/common/meta_service.rs#L62). Instead of previous "127.0.0.1:2333", we use a synthetic hostaddr "risectl-[uuid]", for each run of risectl node. The problem with previous "127.0.0.1:2333" is concurrent risectl nodes sharing the same hostaddr may interference with each other, e.g. unpinning hummock version being used by other.
- [Unregister risectl node after use](https://github.com/risingwavelabs/risingwave/blob/ab326f32de70706eba0d92b8688a6f01e0d2dffb/src/ctl/src/common/meta_service.rs#L65). We add a `CtlContext` as the provider of `meta_client` and `hummock_client`, thus we can release their resource in a unified manner (CtlContext::try_close). Previously we don't unregister risectl node, which causes them holding resources like pinned version unnecessarily until heartbeat timeout.
- Also remove heartbeat thread's shutdown, cause I think killing the ctl process is enough.

## Checklist

- [x] I have written necessary rustdoc comments
~~- [ ] I have added necessary unit tests and integration tests~~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: src/ctl/src/common/context.rs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                );
+            }
+        }
+    }
+}
+
+impl CtlContextInner {
+    async fn get_meta_client(&mut self) -> anyhow::Result<MetaClient> {
+        let meta_client = match &self.meta_client {
+            None => {
```
Comment (BugenZhao) [inline src/ctl/src/common/context.rs:72]: Does [`OnceCell`](https://docs.rs/tokio/latest/tokio/sync/struct.OnceCell.html#method.get_or_try_init) help here?

File: src/ctl/src/common/context.rs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+#[derive(Default)]
+struct CtlContextInner {
+    meta_client: Option<MetaClient>,
+    hummock: Option<(MonitoredStateStore<HummockStorage>, Metrics)>,
+}
+
+impl CtlContext {
+    pub async fn get_meta_client(&self) -> anyhow::Result<MetaClient> {
```
Comment (BugenZhao) [inline src/ctl/src/common/context.rs:34]: We can simply name it `meta_client`.

https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter

```suggestion
    pub async fn meta_client(&self) -> anyhow::Result<MetaClient> {
```

Comment (zwang28) [reply src/ctl/src/common/context.rs]: Fixed, thanks.

File: src/ctl/src/common/context.rs (hunk 1 above)
Comment (zwang28) [reply src/ctl/src/common/context.rs]: Fixed, thanks.

**Review by BugenZhao — APPROVED** [review]
LGTM! 🥰

**Pull request merged**